src/LSST/control/Ilc.h

```diff
@@ -0,0 +1,129 @@
+/*
+ *  This file is part of LSST M2 support system package.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// System headers
+#include <chrono>
+#include <map>
+#include <memory>
+#include <mutex>
+#include <stdint.h>
+#include <string>
+#include <vector>
+
+// Project headers
+#include "control/DaqInMock.h"
+#include "util/Log.h"
+
+#ifndef LSST_M2CELLCPP_CONTROL_ILC_H
+#define LSST_M2CELLCPP_CONTROL_ILC_H
+
+namespace LSST {
+namespace m2cellcpp {
+namespace control {
+
+/// This class contains information about one ILC.
+/// FUTURE: Possibly use this as a base class for Axial and Tangential ILC classes.
+/// Unit tests in test_FpgaIo.cpp
+class Ilc {
+public:
+    using Ptr = std::shared_ptr<Ilc>;
+
+    /// Create a new Ilc
+    Ilc(std::string const& name, int idNum);
+
+    Ilc() = delete;
+    Ilc(Ilc const&) = delete;
+    Ilc& operator=(Ilc const&) = delete;
+
+    ~Ilc() = default;
+
+    /// Return the `_name` of the ILC.
+    std::string getName() { return _name; }
+
+    /// Return the `_idNum` of the ILC.
+    int getIdNum() { return _idNum; }
+
+    /// Return value of `bit` from `byt`.
```

Comment:
byt -> byte??? Or a better name?

Reply:
Changed to "byteIn". C++ 17 added std::byte, and I'd rather not name something after a std type.